src/Database/LSMTree/Internal.hs

```diff
@@ -1105,9 +1105,17 @@ snapshot resolve snap label tableType t = do
     let conf = tableConfig t
     withOpenTable t $ \thEnv -> do
       let hfs = tableHasFS thEnv
-      let snapPath = Paths.snapshot (tableSessionRoot thEnv) snap
-      FS.doesFileExist (tableHasFS thEnv) snapPath >>= \b ->
-              when b $ throwIO (ErrSnapshotExists snap)
+
+      -- Guard that the snapshot does not exist already
+      let snapDir = Paths.namedSnapshotDir (tableSessionRoot thEnv) snap
+      doesSnapshotExist <-
+        FS.doesDirectoryExist (tableHasFS thEnv) (Paths.getNamedSnapshotDir snapDir)
+      if doesSnapshotExist then
+        throwIO (ErrSnapshotExists snap)
+      else
+        -- we assume the snapshots directory already exists, so we just have to
```

Comment:
Where is the snapshots directory created? Should we handle the error when it does not exist?